### PR TITLE
adds gnu-parallel to fix integration test

### PR DESCRIPTION
**Pull request by wgahnagl**
#### What type of PR is this?

/kind ci 

#### What this PR does / why we need it:
fixes the integration tests to report failure properly, and adds gnu-parallel to the prow CI 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: contrib/test/ci/system-packages.yml [hunk 1]
```diff
@@ -58,6 +58,25 @@
       - tar
       - wget
 
+- name: add gnu-parallel
```
Comment (rphillips) [inline contrib/test/ci/system-packages.yml:61]: `parallel` added to the list above should work

File: contrib/test/ci/test.yml [hunk 2]
```diff
@@ -64,9 +68,12 @@
       poll: 30
 
     - name: run integration tests with userNS enabled + alternate results file
-      shell: "make localintegration 2>&1 | tee /tmp/artifacts/testout.txt"
+      shell: | 
+        set -eou pipefail
+        make localintegration 2>&1 | tee /tmp/artifacts/testout.txt
```
Comment (rphillips) [inline contrib/test/ci/test.yml:73]: this should be a different filename so it doesn't stomp on the integration tests in the previous block.

Comment (rphillips) [reply contrib/test/ci/test.yml:73]: make localintegration 2>&1 > /tmp/artifacts/testout.txt

and

make localintegration 2>&1 > /tmp/artifacts/test_with_userns.txt [or something like that]

**Comment by saschagrunert** [comment]
@wgahnagl please dco sign-off your commits :)

**Comment by wgahnagl** [comment]
no longer needed, fixes were merged by other PRs

**Pull request closed without merging**